src/coreComponents/fileIO/logPart/LogPart.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file LogPart.cpp
+ */
+
+#include "LogPart.hpp"
+#include <algorithm>
+
+namespace geos
+{
+
+LogPart::LogPart( string_view logPartTitle ):
+  m_logPartTitle( string( logPartTitle ) ),
+  m_logPartWidth( std::max((size_t)m_rowMinWidth, logPartTitle.length()) )
+{
+  m_footerTitle = GEOS_FMT( "{}", m_logPartTitle );
+}
+
+void LogPart::addDescription( string const & description )
+{
+  m_descriptionNames.push_back( description );
+  m_descriptionValues.push_back( std::vector< string >() );
+}
+
+
+void LogPart::setMinWidth( size_t const & minWidth )
+{
+  m_rowMinWidth = minWidth;
+  m_logPartWidth = std::max( m_rowMinWidth, m_logPartWidth );
+}
+
+void LogPart::formatDescriptions()
+{
+  size_t maxNameSize = 1;
+
+  for( auto const & name : m_descriptionNames )
+  {
+    size_t const idx = &name - &(*m_descriptionNames.begin());
+
+    if( !m_descriptionValues[idx].empty())
+    {
+      maxNameSize = std::max( maxNameSize, name.size() );
+    }
+  }
+
+  size_t const spacesFromBorder = m_borderMargin * 2 + m_nbBorderChar * 2;
+
+  for( size_t idxName = 0; idxName < m_descriptionNames.size(); idxName++ )
+  {
+
+    if( m_descriptionValues[idxName].empty())
+    {
+      m_formattedDescriptions.push_back( m_descriptionNames[idxName] );
+    }
+    else
+    {
+      string_view name = m_descriptionNames[idxName];
+      string_view spaces = std::string( maxNameSize - name.size(), ' ' );
+      string_view nameFormatted = GEOS_FMT( "{}{} ", name, spaces );
+
+      m_formattedDescriptions.push_back( GEOS_FMT( "{}{}", nameFormatted, m_descriptionValues[idxName][0] ));
```

Comment:
I think it would less mask what you want to do.
```suggestion
      string_view spaces = std::string( maxNameSize - name.size(), ' ' );
      string_view nameFormatted = GEOS_FMT( "{}{}", name, spaces );

      m_formattedDescriptions.push_back( GEOS_FMT( "{} {}", nameFormatted, m_descriptionValues[idxName][0] ));
```